docs/WithListContext.md

```diff
@@ -0,0 +1,184 @@
+
+```jsx
+<WithListContext render={({ data }) => (
+    <ul>
+        {data.map(record => (
+            <li key={record.id}>{record.title}</li>
+        ))}
+    </ul>
+)}>
+```
+
+As a reminder, the [`ListContext`](./useListContext.md) is an object with the following properties:
+
+```jsx
+<WithListContext render={({
+    // fetched data
+    data, // an array of the list records, e.g. [{ id: 123, title: 'hello world' }, { ... }]
+    total, // the total number of results for the current filters, excluding pagination. Useful to build the pagination controls, e.g. 23      
+    isLoading, // boolean that is true until the data is available for the first time
+    isFetching, // boolean that is true while the data is being fetched, and false once the data is fetched
+    // pagination
+    page, // the current page. Starts at 1
+    perPage, // the number of results per page. Defaults to 25
+    setPage, // a callback to change the page, e.g. setPage(3)
+    setPerPage, // a callback to change the number of results per page, e.g. setPerPage(25)
+    hasPreviousPage, // boolean, true if the current page is not the first one
+    hasNextPage, // boolean, true if the current page is not the last one
+    // sorting
+    sort, // a sort object { field, order }, e.g. { field: 'date', order: 'DESC' }
+    setSort, // a callback to change the sort, e.g. setSort({ field: 'name', order: 'ASC' })
+    // filtering
+    filterValues, // a dictionary of filter values, e.g. { title: 'lorem', nationality: 'fr' }
+    displayedFilters, // a dictionary of the displayed filters, e.g. { title: true, nationality: true }
+    setFilters, // a callback to update the filters, e.g. setFilters(filters, displayedFilters)
+    showFilter, // a callback to show one of the filters, e.g. showFilter('title', defaultValue)
+    hideFilter, // a callback to hide one of the filters, e.g. hideFilter('title')
+    // record selection
+    selectedIds, // an array listing the ids of the selected rows, e.g. [123, 456]
+    onSelect, // callback to change the list of selected rows, e.g. onSelect([456, 789])
+    onToggleItem, // callback to toggle the selection of a given record based on its id, e.g. onToggleItem(456)
+    onUnselectItems, // callback to clear the selection, e.g. onUnselectItems();
+    // misc
+    defaultTitle, // the translated title based on the resource, e.g. 'Posts'
+    resource, // the resource name, deduced from the location. e.g. 'posts'
+    refetch, // callback for fetching the list data again
+}) => ( ... )}>
+```
+
+## Availability
+
+Whenever you use a react-admin component fetching a list of records, react-admin stores it in a [`ListContext`](./useListContext.md). This means that `<WithListContext>` works out of the box:
+
+- in descendants of the [`<List>`](./ListBase.md), [`<InfiniteList>`](./InfiniteList.md), and [`<ListBase>`](./ListBase.md) components
```

Comment:
```suggestion
- as a child of the [`<List>`](./ListBase.md), [`<InfiniteList>`](./InfiniteList.md), and [`<ListBase>`](./ListBase.md) components
```

Reply:
I find the existing sentence confusing

Reply:
Your suggestion leads to another confusion: that the component would only work when used as child and not as grandchild. 

I think that "descendants" is the right term.